### PR TITLE
Properly parse nested option assignments

### LINKED ISSUE #1541: Wire cannot resolve option definition such as `(package.option).message.value`
For instance, `option (my_message_option_one).baz.value = BAZ;` will crash on linking. See `custom_options.proto`.

### PR DESCRIPTION
**Pull request by oldergod**
fixes #1541

### CODE REVIEW

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/OptionReader.kt [hunk 1]
```diff
@@ -69,7 +69,7 @@ class OptionReader(internal val reader: SyntaxReader) {
     val kindAndValue = readKindAndValue()
     var kind = kindAndValue.kind
     var value = kindAndValue.value
-    if (subName != null) {
+    subNames.reversed().forEach { subName ->
```
Comment (swankjesse) [inline wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/OptionReader.kt:72]: we should talk about forEach() vs. for()

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt [hunk 2]
```diff
@@ -1879,4 +1879,26 @@ class SchemaTest {
...
+  fun nestedOptionSetting() {
+    val schema = RepoBuilder()
+        .add("dinosaur.proto", """
+              |package wire;
+              |import 'google/protobuf/descriptor.proto';
+              |extend google.protobuf.FieldOptions {
+              |  optional Foo my_field_option = 60004;
+              |}
+              |message Foo {
+              |  optional double a = 1 [(wire.my_field_option).baz.value = "b"];
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt:1893]: Nice test

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt [hunk 3]
```diff
@@ -2329,4 +2329,50 @@ class ProtoParserTest {
     )
     assertThat(ProtoParser.parse(location, proto)).isEqualTo(expected)
   }
+
+  @Test
+  fun deepOptionAssignments() {
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2334]: Nice

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/OptionReader.kt (hunk 1 above)
Comment (oldergod) [reply wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/OptionReader.kt:72]: Yes ! I know you're for the latter...

**Pull request merged**